### PR TITLE
Add: new theming API to powerapps-component-framework

### PR DESCRIPTION
**Pull request by lesyk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: docs will be publish shortly after
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/powerapps-component-framework/powerapps-component-framework-tests.ts
1. types/powerapps-component-framework/componentframework.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a2dfbb5a63f662f6893dbb30ae84c974872c74bd
</p>

File: types/powerapps-component-framework/componentframework.d.ts [hunk 1]
```diff
@@ -1640,7 +2357,7 @@ declare namespace ComponentFramework {
 
         /**
          * Entity metadata refer to online documentation
-         * https://learn.microsoft.com/power-apps/developer/model-driven-apps/clientapi/reference/xrm-utility/getentitymetadata
```
Comment (fafuxa-ms) [inline types/powerapps-component-framework/componentframework.d.ts:1643]: I think the old doc link is more correct - the new one goes to `customer-engagement` which I believe is for OnPrem. So I'd revet this line

Comment (fafuxa-ms) [reply types/powerapps-component-framework/componentframework.d.ts:1643]: There are two other link updates I'd revert

Comment (lesyk) [reply types/powerapps-component-framework/componentframework.d.ts]: Reverted.

**Review by jopursle — APPROVED** [review]

**Review by kaushikkaul — APPROVED** [review]

**Comment by kaushikkaul** [comment]
Ready to merge

**Pull request merged**